go/database/mpt/tool/verify_proof.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package main
+
+import (
+	"fmt"
+	"github.com/Fantom-foundation/Carmen/go/common/interrupt"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt/io"
+	"github.com/urfave/cli/v2"
+	"strings"
+)
+
+var VerifyProof = cli.Command{
+	Action:    verifyProof,
+	Name:      "verify-proof",
+	Usage:     "verifies the consistency of witness proofs",
+	ArgsUsage: "<director>",
+	Flags: []cli.Flag{
+		&cpuProfileFlag,
+		&blockTo,
+		&blockFrom,
+		&blockNum,
+	},
+}
+
+var (
+	blockFrom = cli.IntFlag{
+		Name:  "block-from",
+		Usage: "the starting block number to verify",
+		Value: 0,
+	}
+	blockTo = cli.IntFlag{
+		Name:  "block-to",
+		Usage: "the ending block number to verify",
+		Value: 0,
+	}
+	blockNum = cli.IntFlag{
+		Name:  "block-num",
+		Usage: "block number to verify",
+		Value: 0,
+	}
+)
```

Comment:
Do I understand correctly that user can either choose range of blocks using `--blockFrom` and `--blockTo` or a single block `--blockNum`?

If so, is that really necessary? Shouldn't we just use `--blockFrom 1 --blockTo 1` as single block? I think it removes some possible confusions.

Reply:
right, it was meant for user convenience - exactly as you write, it is a shortcut for having to write` --blockFrom 1 --blockTo 1` - this itself is confusing, because I would guess it must have ben written as `--blockFrom 1 --blockTo 2`, i.e the single param variant prevents confusion

Reply:
ic. Lets leave it as it is.